### PR TITLE
vscode: fix vscode-insiders on darwin

### PR DESCRIPTION
**Pull request by tricktron**
###### Description of changes

Building vscode-insiders according to https://nixos.wiki/wiki/Visual_Studio_Code#Insiders_Build fails on darwin because the binary in the darwin vscode-insiders is still called `code` instead of `code-insiders`and thus the symlinking breaks (On linux it is correctly called `code-insiders`).

As this is likely to change again, I also moved the responsible `sourceExecutableName` argument one layer up from generic.nix to vscode.nix to make it more easily overridable in the future.

Lastly, I also set the pname to vscode-insiders if insiders = true for a better integration with home-manager since it uses the package name to set the extensions dir (e.g. ~/.code vs ~/.code-insiders) accordingly.

As a result, vscode-insiders builds with the documented expression and works out of the box if added to home-manager packages as shown below:

```nix
# home-manager config
vscode.package = (pkgs.vscode.override { isInsiders = true; }).overrideAttrs (oldAttrs: rec 
{
    src = (builtins.fetchTarball 
    {
        url    = "https://code.visualstudio.com/sha/download?build=insider&os=darwin-arm64";
        sha256 = "sha256:1cfd9x2shsh664cniysbj0vj5lkhyy7gxd84hl805nf2y9r6m8qs";
    });
    version = "latest";
});
```

###### Things done

- Built on platform(s)
  - [ ] x86_64-linux
  - [ ] aarch64-linux
  - [x] x86_64-darwin
  - [x] aarch64-darwin
- [ ] For non-Linux: Is `sandbox = true` set in `nix.conf`? (See [Nix manual](https://nixos.org/manual/nix/stable/command-ref/conf-file.html))
- [ ] Tested, as applicable:
  - [NixOS test(s)](https://nixos.org/manual/nixos/unstable/index.html#sec-nixos-tests) (look inside [nixos/tests](https://github.com/NixOS/nixpkgs/blob/master/nixos/tests))
  - and/or [package tests](https://nixos.org/manual/nixpkgs/unstable/#sec-package-tests)
  - or, for functions and "core" functionality, tests in [lib/tests](https://github.com/NixOS/nixpkgs/blob/master/lib/tests) or [pkgs/test](https://github.com/NixOS/nixpkgs/blob/master/pkgs/test)
  - made sure NixOS tests are [linked](https://nixos.org/manual/nixpkgs/unstable/#ssec-nixos-tests-linking) to the relevant packages
- [] Tested compilation of all packages that depend on this change using `nix-shell -p nixpkgs-review --run "nixpkgs-review rev HEAD"`. Note: all changes have to be committed, also see [nixpkgs-review usage](https://github.com/Mic92/nixpkgs-review#usage)
- [x] Tested basic functionality of all binary files (usually in `./result/bin/`)
- [23.11 Release Notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2311.section.md) (or backporting [23.05 Release notes](https://github.com/NixOS/nixpkgs/blob/master/nixos/doc/manual/release-notes/rl-2305.section.md))
  - [ ] (Package updates) Added a release notes entry if the change is major or breaking
  - [ ] (Module updates) Added a release notes entry if the change is significant
  - [ ] (Module addition) Added a release notes entry if adding a new NixOS module
- [x] Fits [CONTRIBUTING.md](https://github.com/NixOS/nixpkgs/blob/master/CONTRIBUTING.md).

### CODE REVIEW

**Comment by domenkozar** [comment]
Looks like it breaks eval.

**Comment by tricktron** [comment]
@NixOS/darwin-maintainers Does anyone have time for a review?

File: pkgs/applications/editors/vscode/vscode.nix [hunk 1]
```diff
@@ -5,6 +5,12 @@
 , nixosTests
 , srcOnly
 , isInsiders ? false
+# sourceExecutableName is the name of the binary in the source archive over
+# which we have no control and it is needed to run the insider version as
+# documented in https://nixos.wiki/wiki/Visual_Studio_Code#Insiders_Build
+# On MacOS the insider binary is still called code instead of code-insiders as
+# of 04-07-2023.
```
Comment (emilazy) [inline pkgs/applications/editors/vscode/vscode.nix:12]: Nit: Please use the international/ISO standard `YYYY-MM-DD` format here to be unambiguous.

**Review by emilazy — APPROVED** [review]
I don't use the Nixpkgs `vscode` but this seems like basically a trivial and fine change to me, modulo one little thing.

File: pkgs/applications/editors/vscode/vscode.nix (hunk 1 above)
Comment (tricktron) [reply pkgs/applications/editors/vscode/vscode.nix]: Done😃

**Review by marsam — APPROVED** [review]

**Pull request merged**